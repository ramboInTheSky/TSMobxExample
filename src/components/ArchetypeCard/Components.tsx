import styled from '@emotion/styled'
import { Paper, IconButton } from '@material-ui/core'

import { variables } from '../../constants/theme'

export const Wrapper: React.ComponentType<any> = styled(Paper)`
    margin-top: ${variables.gutter};
    background-color: ${variables.componentBackgroundClear} !important;
`
export const Header = styled.div`
    background-color: ${variables.archetypeColor};
    color: #333;
    font-size: 1.3rem;
    height: 25px;
    padding: ${variables.indentation};
    display: flex;
    justify-content: space-between;
`

export const Content: React.ComponentType<any> = styled.div`
    background-color: ${variables.componentBackgroundClear} !important;
    padding: ${variables.indentation};
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
`

export const CloseButton: React.ComponentType<any> = styled(IconButton)`
        padding: 2px !important;
`

export const LockGroup: React.ComponentType<any> = styled(Paper)`
    background-color: ${variables.componentBackground} !important;
    margin: ${variables.indentation};
    padding: ${variables.indentation};
    padding-right: calc(${variables.indentation} + 6px);
`