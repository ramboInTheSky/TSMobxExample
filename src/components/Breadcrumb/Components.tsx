import styled from '@emotion/styled'
import { variables } from '../../constants/theme'

export const Wrapper = styled.div`
    margin-top: ${variables.indentation};
    font-size: 12px;
    /* font-weight: bold; */
	& svg{
        vertical-align: sub;
    }
`
