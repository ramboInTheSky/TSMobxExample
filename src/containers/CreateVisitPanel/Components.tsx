import styled from '@emotion/styled'
import { variables } from '../../constants/theme';
import { Paper } from '@material-ui/core';
import { Select } from '@material-ui/core';

export const Wrapper = styled.div`

`

export const Container: React.ComponentType<any> = styled(Paper)`
    width: 100%;
    display: flex;
    flex-direction: column;
    /* justify-content: space-evenly;
    padding-top: 3rem;
    padding-bottom: 4rem; */
`

export const TitleField = styled.span`
    min-width: 200px;
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
    flex-wrap: wrap;
`

export const Title = styled.div`
    font-size: ${variables.headerFontSize};
    font-weight: bold;
    padding: ${variables.gutter};
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: space-between;
`
export const ProfileName = styled.span`
    color: ${variables.primaryColor};
    font-weight: bold;
    padding-right: ${variables.indentation};
    /* font-size: 1.5rem; */
    & svg{
        vertical-align: sub;
    }
`

export const PickersContainer: React.ComponentType<any> = styled.div`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    padding-top: 2rem;
    padding-bottom: 4rem;
`

export const DateContainer = styled.div`
    width: 40%;
    display: flex;
    flex-direction: column;
    & div{
        margin-top: ${variables.gutter};
    }
`

export const ArchetypesContainer: React.ComponentType<any> = styled.div`
    padding-top: ${variables.gutter};
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;
    & button{
        min-width: 150px;
        margin: ${variables.indentation};
    }
`

// export const ArchetypesContainer: React.ComponentType<any> = styled.div`
//     padding-top: ${variables.gutter};
//     display: grid;
//     grid-template-columns: 16% 16% 16% 16% 16% 16%;
//     flex-direction: row;
//     justify-content: center;
//     flex-wrap: wrap;
//     & button{
//         width: 95%;
//         min-width: 150px;
//         margin: ${variables.indentation};
//     }
// `

export const Dropdown: React.ComponentType<any> = styled(Select)`
    vertical-align: top;
    width: 250px;
    margin: ${variables.indentation};
`

export const SelectedArchetypesContainer = styled.div`
    margin-top: ${variables.gutter};
`

export const ActionsBar = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	padding: ${variables.gutter};
	/* border-bottom: ${variables.spacer}; */
	padding-bottom: calc(${variables.gutter} * 2);
	padding-top: calc(${variables.gutter} * 2);
    margin-top: calc(${variables.gutter} * 2);
	background-color: ${variables.componentBackgroundClear};
    border-top: 1px solid ${variables.componentBackground};
`
export const ButtonsContainer = styled.div`
	& button {
		margin-left: ${variables.indentation};
	}
`