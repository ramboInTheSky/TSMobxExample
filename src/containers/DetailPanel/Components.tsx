import styled from '@emotion/styled'
import {Paper, TextField} from '@material-ui/core'

import { variables } from '../../constants/theme'

export const Wrapper: React.ComponentType<any> = styled(Paper)`
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	background-color: ${variables.componentBackgroundClear};
	padding: ${variables.gutter};
`

export const Title = styled.div`
	font-size: ${variables.headerFontSize};
	font-weight: bold;
	/* border-bottom: ${variables.spacer};
	line-height: ${variables.lineHeightComponentTitle}; */
`

export const Name = styled.div`
	font-size: 2.2rem;
	font-weight: bold;
`

export const Field = styled.div`
	font-size: ${variables.headerFontSize};
	font-weight: bold;
`

export const FieldsContainer = styled.div`
	color: ${variables.primaryColor};
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	padding: ${variables.indentation};
	margin-top: ${variables.gutter};
	padding-top: ${variables.gutter};
	padding-bottom: ${variables.gutter};
`

export const ActionsBar = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	padding: ${variables.gutter};
	/* border-bottom: ${variables.spacer}; */
	padding-bottom: ${variables.gutter};
	padding-top: calc(${variables.gutter} * 2);
	background-color: ${variables.componentBackgroundClear};
`
export const ButtonsContainer = styled.div`
	& button {
		margin-left: ${variables.indentation};
	}
`

export const InputContainer: React.ComponentType<any> = styled.div`
	display: flex;
	height: 3rem;
	flex-direction: row;
`
export const NotesInputContainer: React.ComponentType<any> = styled.div`
	display: flex;
	height: 8rem;
	flex-direction: row;
`

export const Input: React.ComponentType<any> = styled(TextField)`
	margin-left: 8;
	padding-left: 1rem;
	flex: 1;
`

export const SingleColumnFieldSet = styled.div`
	display: grid;
	grid-template-columns: 100%;
	grid-row-gap: ${variables.gutter};
	padding-top: ${variables.gutter};
`

export const TwoColumnsFieldset = styled.div`
	display: grid;
	grid-template-columns: 49.4% 49.4%;
	grid-column-gap: ${variables.gutter};
	grid-row-gap: ${variables.gutter};
	padding-top: calc(${variables.gutter}*2);
`
