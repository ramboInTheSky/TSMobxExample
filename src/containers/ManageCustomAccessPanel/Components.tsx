import styled from '@emotion/styled';
import { IconButton, Paper, TextField } from '@material-ui/core';
import { variables } from '../../constants/theme';

export const Wrapper = styled.div`

`

export const Container: React.ComponentType<any> = styled(Paper)`
    width: 100%;
    display: flex;
    flex-direction: column;
    /* justify-content: space-evenly;
    padding-top: 3rem;
    padding-bottom: 4rem; */
`

export const TitleField = styled.span`
    min-width: 200px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
`

export const Title = styled.div`
    font-size: ${variables.headerFontSize};
    font-weight: bold;
    padding: ${variables.gutter};
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: space-between;
`
export const ProfileName = styled.span`
    color: ${variables.primaryColor};
    font-weight: bold;
    padding-right: ${variables.indentation};
    /* font-size: 1.5rem; */
    & svg{
        vertical-align: sub;
    }
`

export const AddTagContainer = styled.div`
    display: flex;
    justify-content: center;
    margin: calc(${variables.gutter} * 2);
`

export const InputContainer: React.ComponentType<any> = styled.div`
	display: flex;
	align-items: center;
    width: 60%;
    min-width: 300px;
`

export const Input: React.ComponentType<any> = styled(TextField)`
    margin-left: 8;
    padding-left: 1rem; 
	flex: 1;
`

export const Action: React.ComponentType<any> = styled(IconButton)`padding: 10;`

export const ActionsBar = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: flex-end;
	padding: ${variables.gutter};
	/* border-bottom: ${variables.spacer}; */
	padding-bottom: calc(${variables.gutter} * 2);
	padding-top: calc(${variables.gutter} * 2);
    margin-top: calc(${variables.gutter} * 2);
	background-color: ${variables.componentBackgroundClear};
    border-top: 1px solid ${variables.componentBackground};
`
export const ButtonsContainer = styled.div`
	& button {
		margin-left: ${variables.indentation};
	}
`
export const RoomsContainer = styled.div`
    margin-top: ${variables.gutter};
`

export const ActionIcon: React.ComponentType<any> = styled(IconButton)`
	margin: auto !important;
	padding-top: 2px !important;
`

export const VisitDateFrom = styled.span`
    color: ${variables.visitDateFromColor};
    font-style: italic;
`

export const VisitDateTo = styled.span`
    color: ${variables.visitDateToColor};
    font-style: italic;
`
