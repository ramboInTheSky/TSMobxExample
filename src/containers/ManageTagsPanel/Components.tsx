import styled from '@emotion/styled'
import { variables } from '../../constants/theme';
import Paper from '@material-ui/core/Paper'
import InputBase from '@material-ui/core/InputBase'
import IconButton from '@material-ui/core/IconButton'
import TextField from '@material-ui/core/TextField'

export const Wrapper = styled.div`

`

export const Container: React.ComponentType<any> = styled(Paper)`
    width: 100%;
    display: flex;
    flex-direction: column;
    /* justify-content: space-evenly;
    padding-top: 3rem;
    padding-bottom: 4rem; */
`

export const TitleField = styled.span`
    min-width: 200px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
`

export const Title = styled.div`
    font-size: ${variables.headerFontSize};
    font-weight: bold;
    padding: ${variables.gutter};
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: space-between;
`
export const ProfileName = styled.span`
    color: ${variables.primaryColor};
    font-weight: bold;
    padding-right: ${variables.indentation};
    /* font-size: 1.5rem; */
    & svg{
        vertical-align: sub;
    }
`

export const AddTagContainer = styled.div`
    display: flex;
    justify-content: center;
    margin: calc(${variables.gutter} * 2);
`


export const InputContainer: React.ComponentType<any> = styled.div`
	display: flex;
	align-items: center;
    width: 60%;
    min-width: 300px;
`

export const Input: React.ComponentType<any> = styled(TextField)`
    margin-left: 8;
    padding-left: 1rem; 
	flex: 1;
`

export const Action: React.ComponentType<any> = styled(IconButton)`padding: 10;`

export const ActionsBar = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: flex-end;
	padding: ${variables.gutter};
	/* border-bottom: ${variables.spacer}; */
	padding-bottom: calc(${variables.gutter} * 2);
	padding-top: calc(${variables.gutter} * 2);
    margin-top: calc(${variables.gutter} * 2);
	background-color: ${variables.componentBackgroundClear};
    border-top: 1px solid ${variables.componentBackground};
`
export const ButtonsContainer = styled.div`
	& button {
		margin-left: ${variables.indentation};
	}
`

export const VisitDateFrom = styled.span`
    color: ${variables.visitDateFromColor};
    font-style: italic;
`

export const VisitDateTo = styled.span`
    color: ${variables.visitDateToColor};
    font-style: italic;
`
