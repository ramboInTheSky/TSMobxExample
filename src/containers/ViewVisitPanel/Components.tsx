import styled from '@emotion/styled'
import { variables } from '../../constants/theme';
import { Paper } from '@material-ui/core';
import { Select } from '@material-ui/core';

export const Wrapper = styled.div`

`

export const Container: React.ComponentType<any> = styled(Paper)`
    width: 100%;
    display: flex;
    flex-direction: column;
    /* justify-content: space-evenly;
    padding-top: 3rem;
    padding-bottom: 4rem; */
`

export const TitleField = styled.span`
    min-width: 200px;
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
    flex-wrap: wrap;
`

export const Title = styled.div`
    font-size: ${variables.headerFontSize};
    font-weight: bold;
    padding: ${variables.gutter};
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: space-between;
`
export const ProfileName = styled.span`
    color: ${variables.primaryColor};
    font-weight: bold;
    padding-right: ${variables.indentation};
    /* font-size: 1.5rem; */
    & svg{
        vertical-align: sub;
    }
`

export const DatesContainer: React.ComponentType<any> = styled.span`
    width: 70%;
    display: flex;
    flex-direction: row;
    justify-content: center;
    padding-top: 1rem;
    padding-bottom: 2rem;
    font-size: 2rem;
    margin: auto;
`

export const DateContainer = styled.div`
    width: 40%;
    display: flex;
    flex-direction: column;
    text-align: center;
    color: ${variables.visitDateFromColor};
    & div{
        margin-top: ${variables.gutter};
    }
`

export const Dropdown: React.ComponentType<any> = styled(Select)`
    vertical-align: top;
    width: 250px;
    margin: ${variables.indentation};
`

export const RoomsContainer = styled.div`
    margin-top: ${variables.gutter};
`
export const TagsContainer = styled.div`
    margin-top: ${variables.gutter};
`
export const ArchetypesContainer = styled.div`
    margin-top: ${variables.gutter};
`

export const ActionsBar = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	padding: ${variables.gutter};
	/* border-bottom: ${variables.spacer}; */
	padding-bottom: calc(${variables.gutter} * 2);
	padding-top: calc(${variables.gutter} * 2);
    margin-top: calc(${variables.gutter} * 2);
	background-color: ${variables.componentBackgroundClear};
    border-top: 1px solid ${variables.componentBackground};
`
export const ButtonsContainer = styled.div`
	& button {
		margin-left: ${variables.indentation};
	}
`