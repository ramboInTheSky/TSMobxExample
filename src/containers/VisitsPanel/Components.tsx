import styled from '@emotion/styled'

import { variables } from '../../constants/theme'
import Paper from '@material-ui/core/Paper';

export const Wrapper = styled.div`
	margin-top: ${variables.gutter};
	padding-top: ${variables.gutter};
	display: grid;
	grid-template-columns: 99.4%;
	grid-column-gap: ${variables.gutter};
	grid-row-gap: ${variables.gutter};
	background-color: ${variables.componentBackgroundClear};
`

export const Entry: React.ComponentType<any>  = styled(Paper)`
	display: grid;
	grid-template-columns: 80% 18%;
	grid-column-gap: ${variables.gutter};
	grid-row-gap: ${variables.gutter};
	/* background-color: ${variables.componentBackground}; */
`

export const AccessContainer = styled.div`
	background-color: ${variables.componentBackground};
	padding: ${variables.gutter};
`
export const DatesContainer = styled.div``

export const Dates = styled.span`font-weight: bold;`
export const Times = styled.span`
	font-weight: inherit;
	padding-left: ${variables.gutter};
`

export const Access = styled.div`padding-top: ${variables.indentation};`
export const Status = styled.div`
	background-color: ${variables.componentBackgroundClear};
	margin: auto;
	font-weight: bold;
	color: #666;
	font-size: 1.3rem;
`
