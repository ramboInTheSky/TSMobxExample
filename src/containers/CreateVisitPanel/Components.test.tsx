import React from 'react'
import { shallow } from 'enzyme'

import { Wrapper, Title, PickersContainer, DateContainer, ArchetypesContainer, } from './Components'

describe('<Wrapper />', () => {
	it('renders correctly', () => {
		expect(shallow(<Wrapper />)).toMatchSnapshot()
	})
})

describe('<Title />', () => {
	it('renders correctly', () => {
		expect(shallow(<Title />)).toMatchSnapshot()
	})
})

describe('<PickersContainer />', () => {
	it('renders correctly', () => {
		expect(shallow(<PickersContainer />)).toMatchSnapshot()
	})
})

describe('<DateContainer />', () => {
	it('renders correctly', () => {
		expect(shallow(<DateContainer />)).toMatchSnapshot()
	})
})

describe('<ArchetypesContainer />', () => {
	it('renders correctly', () => {
		expect(shallow(<ArchetypesContainer />)).toMatchSnapshot()
	})
})

