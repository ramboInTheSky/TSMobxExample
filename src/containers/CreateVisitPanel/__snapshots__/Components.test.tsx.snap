// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArchetypesContainer /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<DateContainer /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<PickersContainer /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<Title /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<Wrapper /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
